### ISSUE TITLE
Standard Search: Had to manually restart uwsgi

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/deploy#146]
I think we need to `include: uwsgi` after cloning the Django project, etc., otherwise we reload it before new requirements are installed. Likely affects all Django apps.

/var/log/uwsgi/app/ocds-search.log

```
[normal restart followed by]
Traceback (most recent call last):
  File "./standardsearch/wsgi.py", line 16, in <module>
    application = get_wsgi_application()
  File "/home/standard-search/ocds-search/.ve/lib/python3.6/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
    django.setup(set_prefix=False)
  File "/home/standard-search/ocds-search/.ve/lib/python3.6/site-packages/django/__init__.py", line 19, in setup
    configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
  File "/home/standard-search/ocds-search/.ve/lib/python3.6/site-packages/django/conf/__init__.py", line 79, in __getattr__
    self._setup(name)
  File "/home/standard-search/ocds-search/.ve/lib/python3.6/site-packages/django/conf/__init__.py", line 66, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/standard-search/ocds-search/.ve/lib/python3.6/site-packages/django/conf/__init__.py", line 157, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/home/standard-search/ocds-search/.ve/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "./standardsearch/settings.py", line 58, in <module>
    import sentry_sdk
ModuleNotFoundError: No module named 'sentry_sdk'
Mon May 11 21:15:23 2020 - unable to load app 0 (mountpoint='') (callable not found or import error)
Mon May 11 21:15:23 2020 - *** no app loaded. going in full dynamic mode ***
Mon May 11 21:15:23 2020 - dropping root privileges after application loading
Mon May 11 21:15:23 2020 - *** uWSGI is running in multiple interpreter mode ***
Mon May 11 21:15:23 2020 - gracefully (RE)spawned uWSGI master process (pid: 965)
Mon May 11 21:15:23 2020 - spawned uWSGI worker 1 (pid: 17754, cores: 1)
Mon May 11 21:15:23 2020 - spawned uWSGI worker 2 (pid: 17755, cores: 1)
Mon May 11 21:15:55 2020 - --- no python application found, check your startup logs for errors ---
[pid: 17755|app: -1|req: -1/1] 213.138.111.77 () {54 vars in 1118 bytes} [Mon May 11 21:15:55 2020] GET /v1/search?q=json&base_url=https%3A%2F%2Fstandard.open-contracting.org%2Flatest%2Fen%2F => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (1 switches on core 0)
```

**Comment by Bjwebb** [comment]
We don't see this problem for cove or ocdskingfisherprocess, as we have a explicit uwsgi reload as the last salt step:

https://github.com/open-contracting/deploy/blob/07b393bbba9fade247785973d1529a395858ccf9/salt/cove.sls#L32-L40

**Comment by jpmckinney** [comment]
Instead of that manual reload, wouldn't my proposed solution work?

**Comment by Bjwebb** [comment]
I'm not sure how important that `sleep 10` is to dealing with the Raven issue. (I didn't write it).

**Comment by jpmckinney** [comment]
We no longer use Raven – all has been converted to sentry_sdk.

**Comment by odscjames** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Standard Search: Had to manually restart uwsgi](https://trello.com/c/f55FSzt6/256-standard-search-had-to-manually-restart-uwsgi)

**Comment by odscjames** [comment]
In salt/django.sls we do have "watch_in:  service: uwsgi".

But I notice it's attached to the git state only, and it should also be in the state that sets up virtual environment too

That may explain the "No module named 'sentry_sdk'" error above - if the git was updated, then uwsgi restarted, but pip install had not actually installed sentry_sdk yet.

**Comment by odscjames** [comment]
See above commit: Worked out so if git changes, it runs pip install, restarts UWSGI and runs follow on Django commands (eg migrate). If git not changes, none of that happens.

Tested on a local Virtual machine with Standard Search, then deployed Cove to live for live check.

Also in https://github.com/open-contracting/deploy/commit/ddb3f8a6b830582741558cb5fa1df87364237904 I took the oppertunity to remove the Cove fix Ben highlighted above.
